File: src/main/java/dev/ainterr/minecraft/blockshuffle/gamemodes/DefaultGameMode.java
package dev.ainterr.minecraft.blockshuffle.gamemodes;

import dev.ainterr.minecraft.blockshuffle.PlayerList;
import org.bukkit.entity.Player;

public class DefaultGameMode implements GameMode {
    public void assignBlock(PlayerList players, Player player) {
        players.newBlock(player, null);
    }

    public boolean isRoundOver(PlayerList players) {
        return players.getTotalStatus() == PlayerList.STATUS_SUCCESS;
    }

    @Override
    public void onRoundStart() {

    }
}

File: src/main/java/dev/ainterr/minecraft/blockshuffle/gamemodes/DefaultGameMode.kt
package dev.ainterr.minecraft.blockshuffle.gamemodes

import dev.ainterr.minecraft.blockshuffle.PlayerData
import org.bukkit.entity.Player

class DefaultGameMode : AbstractGameMode() {
    override fun assignBlock(players: PlayerData, player: Player) {
        players.newBlock(player, null)
    }

    override fun isRoundOver(players: PlayerData): Boolean {
        return players.totalStatus == PlayerData.STATUS_SUCCESS
    }
}